System/Metrics/Dimensional.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+module System.Metrics.Dimensional where
+
+import qualified System.Metrics.Counter as Counter
+import qualified System.Metrics.Distribution as Distribution
+import qualified System.Metrics.Gauge as Gauge
+import qualified System.Metrics.Label as Label
+
+import Prelude hiding (lookup)
+import GHC.Exception (Exception)
+import Control.Exception (throwIO)
+import Data.Text (Text)
+import Data.IORef (IORef, atomicModifyIORef', newIORef, readIORef)
+import Data.HashMap.Strict (HashMap, empty)
+import qualified Data.HashMap.Strict as HashMap
+
+type Name = Text
+type Explanation = Text
+type Dimensions = [Text]
+type Point = [Text]
+
+data Dimensional a = Dimensional {
+    _create      :: !(IO a)
```

Comment:
Please stick to the field naming pattern used for other data types (i.e. no leading underscore, but prefixed as needed).